massa-network/src/peer_info_database.rs

```diff
@@ -494,50 +460,35 @@ impl PeerInfoDatabase {
         if !ip.is_global() {
             return Err(NetworkError::InvalidIpError(*ip));
         }
-        let (available_bootstrap_conns, available_nonbootstrap_conns) =
-            self.get_available_out_connection_attempts();
+        let mut update_happened = false;
+        let peer_type = if let Some(p) = self.get_peer_type(ip) {
+            p
+        } else {
+            let mut p = PeerInfo::new(*ip, false);
+            p.active_out_connection_attempts += 1;
+            self.peers.insert(*ip, p);
+            update_happened = true;
+            self.get_peer_type(ip).unwrap()
```

Comment:
comment the unwrap

Reply:
also we have the peer at hand, so you could just do:
* peer_type = peer.type  (get a copy of the peer type)
* insert the peer
* return the peer_type from the scope

That way we don't have unwraps nor useless extra lookups